pkg/controller/topology.go

```diff
@@ -104,6 +106,31 @@ func SupportsTopology(pluginCapabilities rpc.PluginCapabilitySet) bool {
 		utilfeature.DefaultFeatureGate.Enabled(features.Topology)
 }
 
+func topologyKeysLookup(
+	csiNodeLister storagelistersv1.CSINodeLister,
+	selectedNodeName string,
+	driverName string,
+	pvcNodeStore TopologyProvider,
+	pvcUID types.UID) (topologyKeys []string, err error) {
+	selectedCSINode, err := getSelectedCSINode(csiNodeLister, selectedNodeName)
```

Comment:
from discussion in https://github.com/kubernetes-csi/external-provisioner/pull/1413/files#r2377144860, we should read topology keys and node labels from cache first.